Ginger/GingerCoreCommon/VariablesLib/VariableNumber.cs

```diff
@@ -0,0 +1,242 @@
+See the License for the specific language governing permissions and 
+limitations under the License. 
+*/
+#endregion
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using Amdocs.Ginger.Common;
+using Amdocs.Ginger.Common.Enums;
+using Amdocs.Ginger.Repository;
+
+namespace GingerCore.Variables
+{
+    public class VariableNumber : VariableBase
+    {
+        private string mInitialNumberValue;
+
+        private float? mMinValue;
+        private float? mMaxValue;
+
+        private int? mPrecisionValue;
+
+        private bool mIsDecimalValue;
+        
+        [IsSerializedForLocalRepository]
+        public bool IsDecimalValue
+        {
+            set
+            {
+                mIsDecimalValue = value;
+                OnPropertyChanged("IsDecimalValue");
+                OnPropertyChanged("Formula");
+            }
+            get
+            {
+                return mIsDecimalValue;
+            }
+        }
+
+        [IsSerializedForLocalRepository]
+        public float? MinValue
+        {
+            get 
+            {
+                return mMinValue; 
+            }
+            set 
+            {
+                mMinValue = value;
+                OnPropertyChanged("MinValue");
+            }
+        }
+
+        [IsSerializedForLocalRepository]
+        public float? MaxValue
+        {
+            get 
+            { 
+                return mMaxValue;
+            }
+            set
+            { 
+                mMaxValue = value;
+                OnPropertyChanged("MaxValue");
+            }
+        }
+
+        public VariableNumber() { }
+        public override string VariableType
+        {
+            get { return "Number"; }
+        }
+
+        public override string VariableEditPage
+        {
+            get
+            {
+                return "VariableNumberPage";
+            }
+        }
+        public override eImageType Image
+        {
+            get { return eImageType.SequentialExecution; }
+        }
+
+        public override bool SupportResetValue
+        {
+            get { return true; }
+        }
+
+        public override bool SupportAutoValue
+        {
+            get { return false; }
+        }
+
+        public override bool SupportSetValue
+        {
+            get { return true; }
+        }
+
+        public override string VariableUIType
+        {
+            get { return GingerDicser.GetTermResValue(eTermResKey.Variable) + " Number"; }
+        }
+
+        [IsSerializedForLocalRepository]
+        public string InitialNumberValue
+        {
+            set
+            {
+                if(!mIsDecimalValue)
+                {
+                    mInitialNumberValue = GetValidInteger(value).ToString();
+                }
+                else
+                {
+                    mInitialNumberValue = GetFloatWithPrecisionValue(value);
+                }
+
+                Value = value.ToString();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [There's no need to call 'ToString()' on a string.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5355677)